### PR TITLE
unr-5322 view coordinator interface

### PR DESCRIPTION
**Pull request by samiwh**
#### Description
Create an interface wrapping the functionality of the ViewCoordinator to later be used in public interfaces.
Along with some minor change. Moved the order of functions around to be more sensible and removed the non-retry variants of the command senders as they aren't used and don't add extra functionality.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/SpatialView/SpatialOSWorker.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	virtual void SendMetrics(SpatialMetrics Metrics) = 0;
+	virtual void SendLogMessage(Worker_LogLevel Level, const FName& LoggerName, FString Message) = 0;
+
+	virtual Worker_RequestId SendReserveEntityIdsRequest(uint32 NumberOfEntityIds, FRetryData RetryData) = 0;
+	virtual Worker_RequestId SendCreateEntityRequest(TArray<ComponentData> EntityComponents, TOptional<Worker_EntityId> EntityId,
+													 FRetryData RetryData, const FSpatialGDKSpanId& SpanId = {}) = 0;
+	virtual Worker_RequestId SendDeleteEntityRequest(Worker_EntityId EntityId, FRetryData RetryData,
+													 const FSpatialGDKSpanId& SpanId = {}) = 0;
+	virtual Worker_RequestId SendEntityQueryRequest(EntityQuery Query, FRetryData RetryData) = 0;
+	virtual Worker_RequestId SendEntityCommandRequest(Worker_EntityId EntityId, CommandRequest Request, FRetryData RetryData,
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDK/Public/SpatialView/SpatialOSWorker.h:43]: Maybe move it next to the other command-related methods.

File: SpatialGDK/Source/SpatialGDK/Public/SpatialView/ViewCoordinator.h [hunk 2]
```diff
@@ -49,34 +49,47 @@ class ViewCoordinator
...
+	virtual Worker_RequestId SendEntityCommandRequest(Worker_EntityId EntityId, CommandRequest Request, FRetryData RetryData,
+													  const FSpatialGDKSpanId& SpanId = {}) override;
+
+	virtual bool HasEntity(Worker_EntityId EntityId) const override;
+	virtual bool HasComponent(Worker_EntityId EntityId, Worker_ComponentId ComponentId) const override;
+	virtual bool HasAuthority(Worker_EntityId EntityId, Worker_ComponentSetId ComponentSetId) const override;
+
+	Worker_RequestId SendEntityCommandRequest(Worker_EntityId EntityId, CommandRequest Request, TOptional<uint32> TimeoutMillis = {},
+											  const FSpatialGDKSpanId& SpanId = {});
 	Worker_RequestId SendReserveEntityIdsRequest(uint32 NumberOfEntityIds, TOptional<uint32> TimeoutMillis = {});
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDK/Public/SpatialView/ViewCoordinator.h:87]: Are these still needed? They look like a duplicate of the overriden ones.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 3]
```diff
@@ -1167,7 +1167,7 @@ void USpatialNetDriver::Shutdown()
 
 		if (RoutingSystem)
 		{
-			RoutingSystem->Destroy(Connection);
+			RoutingSystem->Destroy(Connection->GetSpatialWorkerInterface());
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:1170]: Need to remove this from the final delivery

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 4]
```diff
@@ -2167,7 +2167,7 @@ void USpatialNetDriver::TickDispatch(float DeltaTime)
 
 		if (RoutingSystem.IsValid())
 		{
-			RoutingSystem->Advance(Connection);
+			RoutingSystem->Advance(Connection->GetSpatialWorkerInterface());
```
Comment (ImprobableNic) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:2170]: This as well

File: SpatialGDK/Source/SpatialGDK/Public/SpatialView/ViewCoordinator.h (hunk 2 above)
Comment (samiwh) [reply SpatialGDK/Source/SpatialGDK/Public/SpatialView/ViewCoordinator.h:87]: They aren't quite as they don't have retry data and skip that process, however this isn't obvious from the api and I can't think where we'd use them. So maybe we just delete them.

**Review by ImprobableNic — APPROVED** [review]

**Review by improbable-dmitrii — APPROVED** [review]
Awesome, glad to have this.

**Pull request merged**